spec/unit/provider/user/directoryservice_spec.rb

```diff
@@ -1021,10 +1021,10 @@ module Puppet::Util::Plist
   describe '#set_shadow_hash_data' do
     let(:users_plist) { {'ShadowHashData' => ['string_data'] } }
 
-    it 'should flush the plist data to disk on OS X < 10.15' do
-      allow(provider.class).to receive(:get_os_version).and_return('10.12')
+    it 'should flush the plist data a temporary file on OS X < 10.15' do
+      allow(provider.class).to receive(:get_os_version).and_return('10.15')
 
-      expect(provider).to receive(:write_users_plist_to_disk)
+      expect(provider).to receive(:write_and_import_shadow_hash_data)
       provider.set_shadow_hash_data(users_plist, pbkdf2_embedded_plist)
     end
 
```

Comment:
I think we can drop the conditional testing here, especially since we don't call `get_os_version` anymore in `set_shadow_hash_data`, and keep a single test for this behavior.